consensus/src/chained_bft/safety/safety_rules.rs

```diff
@@ -261,7 +239,7 @@ impl<T: Payload> SafetyRules<T> {
     /// sent).
     /// Requires that all the ancestors of the block are available for at least up to the last
     /// committed block, might panic otherwise.
-    pub fn voting_rule(
+    pub fn voting_rule<T: Payload>(
```

Comment:
It's interesting that in today's world we don't even need to receive the Block<T> here, but in future we might want to verify the `id` of a block corresponding to the other fields, hence let's keep it here for now (this is going to be the only function with a `<T>` parameter).